app/graphql/mutations/profile/associate_rules.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module Mutations
+  module Profile
+    # Mutation to associate systems with a profile
+    class AssociateRules < BaseMutation
+      graphql_name 'associateRules'
+
+      argument :id, ID, required: true
+      argument :rule_ids, [ID], required: true
+      field :profile, Types::Profile, null: true
+
+      def resolve(args = {})
+        profile = find_profile(args[:id])
+        if profile
+          profile.update_rules(ids: args[:rule_ids])
+          audit_mutation(profile)
```

Comment:
`update_rules` returns number of added and removed rules. Could it be used as a condition for the audit logging? The numbers might be included within the message.

Reply:
Just making sure that I understand :thinking: you want me to emit an audit message like this:
```
Updated N rule associations of profile #{profile.id}
```
?

Reply:
Yes. Check out profiles controller for inspiration. It logs smth similar.